### PR TITLE
General static analysis improvements and Removal of Interop\Container

### PR DESCRIPTION
**Pull request by gsteel**
|    Q          |   A
|-------------- | ------
| Documentation | yes
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | yes

Note that `laminas-servicemanager` is not a hard dependency here, but nothing has changed that might cause issues with v2. 🤞

### CODE REVIEW

File: src/Translator/LoaderPluginManagerFactory.php [hunk 1]
```diff
@@ -2,10 +2,10 @@
 
 namespace Laminas\I18n\Translator;
 
-use Interop\Container\ContainerInterface;
 use Laminas\ServiceManager\Config;
 use Laminas\ServiceManager\FactoryInterface;
 use Laminas\ServiceManager\ServiceLocatorInterface;
+use Psr\Container\ContainerInterface;
```
Comment (Ocramius) [inline src/Translator/LoaderPluginManagerFactory.php:8]: I think this requires a conflict with earlier ServiceManager versions?

Comment (gsteel) [reply src/Translator/LoaderPluginManagerFactory.php:8]: That's probably anything prior to 3.14 right?

File: src/Translator/LoaderPluginManager.php [hunk 2]
```diff
@@ -100,20 +102,20 @@ class LoaderPluginManager extends AbstractPluginManager
...
             return;
         }
 
         throw new InvalidServiceException(sprintf(
-            'Plugin of type %s is invalid; must implement %s\Loader\FileLoaderInterface '
-            . 'or %s\Loader\RemoteLoaderInterface',
+            'Plugin of type %s is invalid; must implement %s or %s',
             is_object($plugin) ? get_class($plugin) : gettype($plugin),
-            __NAMESPACE__,
-            __NAMESPACE__
```
Comment (Ocramius) [inline src/Translator/LoaderPluginManager.php:116]: Interesting how these were completely wrong :O

File: src/Translator/LoaderPluginManagerFactory.php (hunk 1 above)
Comment (Ocramius) [reply src/Translator/LoaderPluginManagerFactory.php:8]: Hmm, thinking that this change is a BC break **unless** we require the servicemanager component (which declares the interop aliases, therefore providing a BC layer)

Comment (gsteel) [reply src/Translator/LoaderPluginManagerFactory.php:8]: It makes sense to me to make it a hard dependency due to the plugin manager. Just say the word and I'll revert or add the dependency

Comment (Ocramius) [reply src/Translator/LoaderPluginManagerFactory.php:8]: I think adding the dependency makes more sense anyway

**Review by Ocramius — APPROVED** [review]

**Comment by Ocramius** [comment]
Thanks @gsteel!

**Pull request merged**